src/NULS/Signer.cpp

```diff
@@ -0,0 +1,166 @@
+// Copyright © 2017-2019 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "Signer.h"
+
+#include "../Hash.h"
+#include "../HexCoding.h"
+#include "../PrivateKey.h"
+#include "../BinaryCoding.h"
+#include "Address.h"
+
+
+
+using namespace TW;
+using namespace TW::NULS;
+
+static const uint64_t MIN_PRICE_PRE_1024_BYTES = 100000;
+
+static inline uint64_t calculateFee(uint64_t size) {
+  uint64_t fee = (size / 1024) * MIN_PRICE_PRE_1024_BYTES;
+  if (size % 1024 > 0) {
+    fee += MIN_PRICE_PRE_1024_BYTES;
+  }
+  return fee;
+}
+
+/// Encodes a 48-bit little-endian value into the provided buffer.
+static inline void encode48LE(uint64_t val, std::vector<uint8_t> &data) {
+  data.push_back(static_cast<uint8_t>(val));
+  data.push_back(static_cast<uint8_t>((val >> 8)));
+  data.push_back(static_cast<uint8_t>((val >> 16)));
+  data.push_back(static_cast<uint8_t>((val >> 24)));
+  data.push_back(static_cast<uint8_t>((val >> 32)));
+  data.push_back(static_cast<uint8_t>((val >> 40)));
+}
+
+/// Encodes a VarInt little-endian value into the provided buffer.
+static inline void encodeVarIntLE(uint64_t val, std::vector<uint8_t> &data) {
+  if (val < 0xFD) {
+    data.push_back(static_cast<uint8_t>(val));
+  } else if (val <= 0xFFFF) {
+    data.push_back(static_cast<uint8_t>(0xFD));
+    data.push_back(static_cast<uint8_t>(val));
+    data.push_back(static_cast<uint8_t>((val >> 8)));
+  } else if (val <= 0xFFFFFFFF) {
+    data.push_back(static_cast<uint8_t>(0xFE));
+    data.push_back(static_cast<uint8_t>(val));
+    data.push_back(static_cast<uint8_t>((val >> 8)));
+    data.push_back(static_cast<uint8_t>((val >> 16)));
+    data.push_back(static_cast<uint8_t>((val >> 24)));
+  } else if (val > 0xFFFFFFFF) {
+    data.push_back(static_cast<uint8_t>(0xFF));
+    data.push_back(static_cast<uint8_t>(val));
+    data.push_back(static_cast<uint8_t>((val >> 8)));
+    data.push_back(static_cast<uint8_t>((val >> 16)));
+    data.push_back(static_cast<uint8_t>((val >> 24)));
+    data.push_back(static_cast<uint8_t>((val >> 32)));
+    data.push_back(static_cast<uint8_t>((val >> 40)));
+    data.push_back(static_cast<uint8_t>((val >> 48)));
+    data.push_back(static_cast<uint8_t>((val >> 56)));
+  }
+}
+
+static inline void serializerRemark(std::string remark, std::vector<uint8_t> &data) {
```

Comment:
Please use constant references to avoid unnecessary copies: `const std::string&`